packages/test/test-end-to-end-tests/src/test/leader.spec.ts

```diff
@@ -1,248 +0,0 @@
-/*!
- * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
- * Licensed under the MIT License.
- */
-
-import { strict as assert } from "assert";
-import { IAgentScheduler, TaskSubscription } from "@fluidframework/agent-scheduler";
-import { Container } from "@fluidframework/container-loader";
-import { requestFluidObject } from "@fluidframework/runtime-utils";
-import { ITestObjectProvider, ITestFluidObject, timeoutPromise } from "@fluidframework/test-utils";
-import { describeFullCompat } from "@fluidframework/test-version-utils";
-
-async function ensureConnected(container: Container) {
-    if (!container.connected) {
-        await timeoutPromise((resolve, rejected) => container.on("connected", resolve), { durationMs: 4000 });
-    }
-}
-
-async function getLeadershipSubscriptionForContainer(container: Container) {
-    const globalScheduler = await requestFluidObject<IAgentScheduler>(container, "_scheduler");
-    const taskSubscription = new TaskSubscription(globalScheduler, "leader");
-    taskSubscription.volunteer();
-    return taskSubscription;
-}
-
-describeFullCompat("Leader", (getTestObjectProvider) => {
```

Comment:
isn't this all still useful coverage? it just now needs to test an instance of the agent scheduler datastore rather than the global instance?

Reply:
Now that the flag is gone it's duplicative with agentScheduler.spec.ts (the "leader" task isn't special since the APIs for leadership are already gone).

Reply:
I suppose it's not a perfect overlap -- I can look at merging some of the other coverage in.

Reply:
ahhh. ok. i didn't realize we had basically duplicate coverage. if we can salvage anything not already covered that would be great.

Reply:
Merged in the state transition tests in latest, which were the non-overlapping bit.